utils/math/src/lib.rs

```diff
@@ -0,0 +1,28 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use num_bigint::BigInt;
+
+pub const PRECISION: u64 = 128;
+
+pub fn poly_val(poly: &[BigInt], x: &BigInt) -> BigInt {
+    let mut res = BigInt::default();
+
+    for coeff in poly {
+        let temp = &res * x;
+        res = (temp >> PRECISION) + coeff;
```

Comment:
```suggestion
        res = ((res * x) >> PRECISION) + coeff;
```
No need for the temp value